docs/influence-sensor-execution-order.md

```diff
@@ -0,0 +1,62 @@
+# Influencing the Sensors execution order
+
+In Sonar products, Sensors are executed one after another. In some situations, it can be useful for a Sensor to always be executed after another one; one use case is that one Sensor builds an intermediate representation of the code, and other Sensors can compute issues from it. There are different mechanisms in the API for plugins to influence the execution order.
+
+## Registering for an execution phase
+
+Sensors are executed by products during a `Phase`. There are 3 of them:
+
+* PRE
+* DEFAULT
+* POST
+
+The phases execute in this order: PRE -> DEFAULT -> POST.
+
+Each Sensor can register for a specific phase. By default, they belong to the DEFAULT phase. This can be changed by annotating the Sensor class with the `Phase` annotation, as follows:
+
+```
+import org.sonar.api.batch.Phase;
+import org.sonar.api.batch.sensor.Sensor;
+
+
+@Phase(name = Phase.Name.PRE)
+public class MySensor implements Sensor {
+
+}
+```
+
+As Phases are executed in a predefined order, registering Sensors in different phases enforces their order of execution.
+
+## Ordering Sensors within a phase
+
+Within a phase, sensors are executed in two steps: first the non-global sensors, then the global ones. A Sensor is marked global by calling `org.sonar.api.batch.sensor.SensorDescriptor#global()` from its `describe` method.
```

Comment:
This is only true for the "root" module. And you forgot to mention `ProjectSensor` that will be executed separately (it is considered as a separate extension point, so they will run after other Sensors, always).

Reply:
You're right. 
Let me see how I can turn this, I want to avoid explaining about modules vs project, and also the different types of sensors, which could be a whole different doc page.